openedx/frontend-app-library-authoring
# feat: add export library tags menu [FC-0049]

## Description
This PR adds an item in the `Settings` menu to allow a call of the function that exports the library tags to a CSV file.

![image](https://github.com/openedx/frontend-app-library-authoring/assets/849463/b55d0c5d-6fb4-4dd8-be14-0441a20c8567)

## Supporting Information
* Closes:
  * https://github.com/openedx/modular-learning/issues/173
* Depends on:
  * https://github.com/openedx/edx-platform/pull/34246

## Testing instructions
1. Use https://github.com/open-craft/taxonomy-sample-data/ to create some taxonomies.
1. Go to the library page in Studio and add some tags to the units in the course using the `Manage Tags` option
![image](https://github.com/openedx/frontend-app-library-authoring/assets/849463/cd9b3f7c-a270-415e-94e1-b87155aa40cc)
1. Checkout **THIS** branch and reload the library authoring page
1. Click on the `Export Tags` menu from the detail page of an existing library
1. Check if a CSV file is downloaded with the Library structure and the tags applied to its items.

---
Private-ref:
 - [FAL-3611](https://tasks.opencraft.com/browse/FAL-3611)
